pkg/asset/installconfig/password.go

```diff
@@ -0,0 +1,63 @@
+package installconfig
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+
+	"github.com/openshift/installer/pkg/asset"
+	"golang.org/x/crypto/ssh/terminal"
+)
+
+var (
+	passwordPrompt = fmt.Sprintf("Enter password (6-16 characters):")
+	reEnterPrompt  = fmt.Sprintf("Re-enter password:")
+)
+
+// password is an asset that queries the user for the password to the cluster
+//
+// Contents[0] is the actual string form of the password
+type password struct {
+	InputReader *bufio.Reader
+}
+
+var _ asset.Asset = (*password)(nil)
+
+// Dependencies returns no dependencies.
+func (a *password) Dependencies() []asset.Asset {
+	return []asset.Asset{}
+}
+
+// Generate queries for input from the user.
+func (a *password) Generate(map[asset.Asset]*asset.State) (*asset.State, error) {
+	password, err := a.queryUserForPassword()
+	return assetStateForStringContents(password), err
+}
+
+// queryUserForPassword for prompts Stdin for password without echoing anything on screen
+// TODO: mask the password characters
```

Comment:
Doesn't `ReadPassword` already do this?

Reply:
ReadPassword hides the echo. By masking I meant putting '*' or something. ReadPassword does not do that by itself. Can add that later.

Reply:
The best library I could find for that has limitations :). So its going to be a cgo thing in a subsequent PR soon. Meanwhile, letting the code be.

Reply:
> So its going to be a cgo thing in a subsequent PR soon.

I don't think it's worth using `cgo` (would we then be distributing the linked C libraries in the installer tarball?).  Also, see #243 where @steveeJ was having *local* linking issues for the smoke tests.  So while `*` characters would be nice, I'm fine just leaving the echo off (like lots of other password-input utilities do; `sudo` comes to mind).

Reply:
Yeah, I'm vetoing `cgo`.